src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -650,49 +669,12 @@ protected final void afterInternal(boolean afterClass) throws Exception {
         } finally {
             if (!success) {
                 // if we failed here that means that something broke horribly so we should clear all clusters
-                afterTestRule.forceFailure();
+                // TODO: just let the exception happen, WTF is all this horseshit
+                // afterTestRule.forceFailure();
             }
         }
     }
 
-    @Override
-    protected final AfterTestRule.Task afterTestTask() {
-        return new AfterTestRule.Task() {
-            @Override
-            public void onTestFailed () {
-                //we can't clear clusters after failure when using suite scoped tests, as we would need to call again
```

Comment:
Does this have a new home (I might have missed it). I believe Luca added this to make sure that shared cluster a reset after a failure because it may have residues...